### ISSUE TITLE
EFLOW and shared folders not triggering inotify events

### ISSUE THREAD

**Issue by beckerben** [Azure/iotedge-eflow#270]
**Question Summary**
Do shared folders support seeing inotify events in EFLOW VM?

**Detailed background**
We were testing with EFLOW for some of our modules and one of them is a file processing based module.  It heavily relies on inotify events in order to determine when to process a file.  This works great on our Linux based hosts and we saw the host file sharing feature was supported for EFLOW so assumed it would work but upon actual testing we are seeing it is not working.  

It looks like any creation or modify events within files from the host in the shared folder are not triggering inotify events in the module container running within EFLOW.

If we connect-eflowvm and then "touch" the file, then it triggers our module to see it, but doing the same thing in the windows host does not.

Furthermore we found some other articles on the net that suggest this may be a limitation of virtio.

[Ref 1](https://stackoverflow.com/questions/32726587/programmatically-trigger-an-inotify-event-on-linux)
[Ref 2](https://github.com/lima-vm/lima/issues/615)

I just wanted to confirm our theory that there is no way to make EFLOW VM see the inotify events for items in shared folders mounted into EFLOW via the [shared folder feature](https://learn.microsoft.com/en-us/azure/iot-edge/how-to-share-windows-folder-to-vm?view=iotedge-1.4)

**Comment by fcabrera23** [comment]
Hi @beckerben,

Thanks for reaching out.  The current file/folder sharing mechanism does not support a way to notify when there are file changes. This is a current limitation of the underlying sharing technology we use.

Thanks,
Francisco

**Comment by beckerben** [comment]
Thanks @fcabrera23, that was the conclusion we had come to so thanks for confirming.

We ended up solving this issue going a different direction with a custom samba module we developed which allows us to mount a samba share on the host and share files via a shared docker volume with access to the inotify events as expected by other modules running within EFLOW VM.   This works well for our scenario where we'd be using EFLOW so we have a path forward which is good!
